includes/login-register.php

```diff
@@ -136,15 +136,16 @@ function edd_log_user_in( $user_id, $user_login, $user_pass, $remember = false )
 		'remember'      => $remember,
 	);
 
-	$user = wp_signon( $credentials );
+	$user     = wp_signon( $credentials );
+	$redirect = $_SERVER['HTTP_REFERER'];
```

Comment:
Let's use `wp_get_referer()` instead. Also, add in a check if it's empty (it sometimes can be) and if so, maybe then we should fall back to `edd_get-checkout_uri()`.